apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java

```
@@ -122,62 +122,88 @@ public String retrieveFDADrugCodeDisplay(Optional<String> claimDrugCode) {
-                ndcProductColumns[3] + " - " + ndcProductColumns[13]);
-          } catch (StringIndexOutOfBoundsException e) {
-            continue;
-          }
+  protected Map<String, String> getFdaProcessedData(
+      boolean includeFakeDrugCode, InputStream inputStream) {
+
+    Map<String, String> ndcProcessedData = new HashMap<String, String>();
+
+    try (final InputStream ndcProductStream = inputStream;
+        final BufferedReader ndcProductsIn =
+            new BufferedReader(new InputStreamReader(ndcProductStream))) {
+      /*
+       * We want to extract the PRODUCTNDC and PROPRIETARYNAME/SUBSTANCENAME from the FDA Products
+       * file (fda_products_utf8.tsv is in /target/classes directory) and put in a Map for easy
+       * retrieval to get the display value which is a combination of PROPRIETARYNAME &
+       * SUBSTANCENAME
+       */
+      String line = "";
+      ndcProductsIn.readLine();
+      while ((line = ndcProductsIn.readLine()) != null) {
+        String ndcProductColumns[] = line.split("\t");
+        try {
+          String nationalDrugCodeManufacturer =
+              StringUtils.leftPad(
+                  ndcProductColumns[1].substring(0, ndcProductColumns[1].indexOf("-")), 5, '0');
+          String nationalDrugCodeIngredient =
+              StringUtils.leftPad(
+                  ndcProductColumns[1].substring(
+                      ndcProductColumns[1].indexOf("-") + 1, ndcProductColumns[1].length()),
+                  4,
+                  '0');
+          // ndcProductColumns[3] - Proprietary Name
+          // ndcProductColumns[13] - Substance Name
+          ndcProcessedData.put(
+              String.format("%s-%s", nationalDrugCodeManufacturer, nationalDrugCodeIngredient),
+              ndcProductColumns[3].replace("\"", "")
+                  + " - "
+                  + ndcProductColumns[13].replace("\"", ""));
+        } catch (StringIndexOutOfBoundsException e) {
+          continue;
```

Why do we ignore this exception?

---

took out the exception